### ISSUE TITLE
Fail to reproduce DSVT-P on nuscenes dataset.

### ISSUE THREAD

**Issue by synsin0** [Haiyang-W/DSVT#55]
Hi, I'd like to reproduce DSVT-P on nuScenes dataset. I use the config you provided. I first eval the checkpoint and config, and gets the correct results, that is mAP=66; NDS=70. Then I train the same config with 8x A6000 GPUs with batch_size=4 for 20 epochs with cbgs. I can only get mAP=55; NDS=63 with checkpoint_20.pth. I wonder what's going wrong with my training process. Do I need more epochs?

**Comment by Haiyang-W** [comment]
The same config is ok.

**Comment by synsin0** [comment]
I compared mine and your provided log. I forgot to add sync_bn as an argument! I have added it and restart my experiment. Hope  for next reproduction. sync_bn may affect a lot.

**Comment by Haiyang-W** [comment]
Nice!  You can try it first. If you have any questions, please feel free to contact with me.

**Comment by Haiyang-W** [comment]
> Note that you should apply the fade strategy in your training process (turn off the gt sampling for the last 4 or 5 epochs).

It has been written into [config](https://github.com/Haiyang-W/DSVT/blob/fa37ee29ddcc662b88e68b42cf03b91dec41ba88/tools/cfgs/dsvt_models/dsvt_plain_1f_onestage_nusences.yaml#L200), so you just run the provided config directly.

**Comment by synsin0** [comment]
After turning on sync_bn, I still can not reproduce the reported metrics. I provide my training log so that you may find possible errors.
[train_20230829-170810.log](https://github.com/Haiyang-W/DSVT/files/12474355/train_20230829-170810.log)

**Comment by Haiyang-W** [comment]
> After turning on sync_bn, I still can not reproduce the reported metrics. I provide my training log so that you may find possible errors.
> 
> [train_20230829-170810.log](https://github.com/Haiyang-W/DSVT/files/12474371/train_20230829-170810.log)

It seems from the log that you are not using dsvt's repo, but pcdet's latest official repo? They may have some differences in training.  Could you completely use dsvt's repo to replicate?

**Comment by Haiyang-W** [comment]
By the way, could you use our repo to preprocess the nuscenes data? I'm not sure what's the difference between our nuscenes and the official pcdet? I will check it laterly.

**Comment by Haiyang-W** [comment]
If anyone can reproduce nuscenes please let me know in this issue.

**Comment by synsin0** [comment]
I start to setup a new DSVT original repo and preprocess and train with default configs. Thanks for your patience.

**Comment by Haiyang-W** [comment]
> I start to setup a new DSVT original repo and preprocess and train with default configs. Thanks for your patience.

If you face some problem, please email me, so that we can look for problems together.

**Comment by synsin0** [comment]
Here is my new training log. I strictly follow the DSVT repo. May you take a look at why my log has only half of the training loss as yours.
[log_train_20230831-141035.txt](https://github.com/Haiyang-W/DSVT/files/12482948/log_train_20230831-141035.txt)

**Comment by chenshi3** [comment]
> Here is my new training log. I strictly follow the DSVT repo. May you take a look at why my log has only half of the training loss as yours. [log_train_20230831-141035.txt](https://github.com/Haiyang-W/DSVT/files/12482948/log_train_20230831-141035.txt)

After the upcoming deadline, which is just two days away, I will review your log and code to identify and address any potential issues or errors.

**Comment by Haiyang-W** [comment]
> Here is my new training log. I strictly follow the DSVT repo. May you take a look at why my log has only half of the training loss as yours. [log_train_20230831-141035.txt](https://github.com/Haiyang-W/DSVT/files/12482948/log_train_20230831-141035.txt)

The print log interval we choose is 1000, So you can run this a little bit longer, or you can restart the program and set the print interval to 1000 as well.

**Comment by synsin0** [comment]
I restart and get similar loss at the first interval=1000, that is loss=9.1516, which is similar to your log loss=9.6196. I will keep track on this experiment to see whether it can exceed NDS=70. If yes, I will close this issue.

**Comment by Haiyang-W** [comment]
> I restart and get similar loss at the first interval=1000, that is loss=9.1516, which is similar to your log loss=9.6196. I will keep track on this experiment to see whether it can exceed NDS=70. If yes, I will close this issue.

Great! We reproduced this code before release (two month ago), and there should be no problem. We hope you can reproduce it. 

Looking forward to your good news, we will also run again soon. :)

**Comment by synsin0** [comment]
My final reproduced result is mAP=0.6668, NDS=0.7122 for checkpoint_20th. Thanks for your help of getting correct results!

**Comment by Haiyang-W** [comment]
> My final reproduced result is mAP=0.6668, NDS=0.7122 for checkpoint_20th. Thanks for your help of getting correct results!

Great! Thanks for your timely reply.

**Comment by Byte247** [comment]
> My final reproduced result is mAP=0.6668, NDS=0.7122 for checkpoint_20th. Thanks for your help of getting correct results!

What were your changes to get to these results? I can only achieve mAP: 0.6224 and NDS: 0.6780
